lib/rules/no-unused-expressions.js

```diff
@@ -129,12 +129,37 @@ module.exports = {
 			/**
 			 * https://tc39.es/ecma262/#directive-prologue
 			 *
-			 * Only `FunctionBody`, `ScriptBody` and `ModuleBody` can have directive prologue.
-			 * Class static blocks do not have directive prologue.
+			 * Only `FunctionBody`, `ScriptBody`, `ModuleBody`, and TypeScript module/namespace bodies
+			 * can have directive prologue. Class static blocks do not have directive prologue.
 			 */
+			if (!looksLikeDirective(node)) {
+				return false;
+			}
+
+			const parent = node.parent;
+			if (!parent || !parent.body) {
+				return false;
+			}
```

Comment:
[Refactor] `looksLikeDirective` filters to just expression statements, which always have a parent with a body. This doesn't fail any unit tests:

```suggestion
			const parent = node.parent;
```

For this and my other "dead code" notes: if I'm wrong, we should have unit tests proving it.

Reply:
Got it, and yes all expression statements will always have a parent with a body, applying your suggestion